Tests/FluentAssertions.Specs/Types/MethodInfoSelectorSpecs.cs

```diff
@@ -272,6 +273,32 @@ public void When_selecting_methods_not_decorated_with_a_noninheritable_attribute
             methods.Should().ContainSingle();
         }
 
+        [Fact]
+        public void When_selecting_methods_that_are_async_it_should_only_return_the_applicable_methods()
+        {
+            // Arrange
+            Type type = typeof(TestClassForMethodSelectorWithAsync);
+
+            // Act
+            IEnumerable<MethodInfo> methods = type.Methods().ThatAreAsync().ToArray();
+
+            // Assert
+            methods.Should().ContainSingle();
+        }
+
+        [Fact]
+        public void When_selecting_methods_that_are_not_async_it_should_only_return_the_applicable_methods()
+        {
+            // Arrange
+            Type type = typeof(TestClassForMethodSelector);
+
+            // Act
+            MethodInfo[] methods = type.Methods().ThatAreNotAsync().ToArray();
+
+            // Assert
+            methods.Length.Should().Be(7);
```

Comment:
`methods.Should().HaveCount(7);`

Reply:
The `7` seems to be a bit of magic number, especially because `TestClassForMethodSelector` is not immediately following the test. Maybe the test itself should use Reflection to calculate the number of expected non-async methods.

Reply:
I'm changing the new class to have one method for each.